src/Glpi/Assets/ImportMapGenerator.php

```diff
@@ -0,0 +1,194 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Assets;
+
+use Glpi\Application\Environment;
+use Plugin;
+use Psr\SimpleCache\CacheInterface;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use Symfony\Component\Filesystem\Path;
+
+/**
+ * Generates an import map for JavaScript modules with cache busting parameters
+ *
+ * @final
+ */
+class ImportMapGenerator
+{
+    /**
+     * @var string
+     */
+    private $root_doc;
+
+    /**
+     * @var string
+     */
+    private $glpi_root;
+
+    /**
+     * @var CacheInterface|null
+     */
+    private $cache;
+
+    /**
+     * @param string $root_doc Root document URL path
+     * @param CacheInterface|null $cache Optional cache instance
+     */
+    public function __construct(string $root_doc, string $glpi_root, ?CacheInterface $cache = null)
+    {
+        $this->root_doc  = $root_doc;
+        $this->glpi_root = $glpi_root;
+        $this->cache     = $cache;
+    }
+
+    /**
+     * Get the list of active plugins
+     *
+     * @return array Array of plugin names
+     */
+    protected function getPluginDirList(): array
+    {
+        return array_map(
+            fn($plugin_key) => Plugin::getPhpDir($plugin_key, true),
+            Plugin::getPlugins()
+        );
+    }
+
+    /**
+     * Generate the import map data
+     *
+     * @return array{imports: array<string, string>} The import map data with module names as keys and URLs as values
+     */
+    public function generate(): array
+    {
+        $should_use_cache = $this->cache !== null && !Environment::get()->shouldExpectResourcesToChange();
+
+        if ($should_use_cache) {
+            $cached_map = $this->cache->get('js_import_map');
+            if ($cached_map !== null) {
+                return $cached_map;
+            }
+        }
+
+        $import_map = [
+            'imports' => [],
+        ];
+
+        // Scan GLPI core directories
+        $this->addModulesToImportMap($import_map, $this->glpi_root . '/js/modules');
+        $this->addModulesToImportMap($import_map, $this->glpi_root . '/public/lib');
+        $this->addModulesToImportMap($import_map, $this->glpi_root . '/public/build');
+
+        // Scan plugin directories
+        foreach ($this->getPluginDirList() as $plugin_dir) {
+            // Check all the possible module directories for plugins
+            $plugin_module_dirs = [
+                $plugin_dir . '/js/modules',
+                $plugin_dir . '/public/lib',
+                $plugin_dir . '/public/build',
```

Comment:
Maybe we can scan all the public dir of active plugins. They are not supposed to contains more than a few JS files.

```suggestion
                $plugin_dir . '/public',
```


Another solution would be to delegate the responsability to register the modules to handle to plugins. For instance with a `ImportMapGenerator::registerModulesPath(string $plugin_key, string $path) :void` method (`ImportMapGenerator::getInstance()->registerModulesPath('myplugin', '/js/modules')`).
It is probably safer as it may prevent unexpected performances issues if a plugin bundles a bunch of unused JS files in its public dir (for instance a plugin that bundles all its node_modules dir due to a bad packaging script).